app/uk/gov/hmrc/mtdsatestsupportapi/models/request/createTestBusiness/Business.scala

```diff
@@ -22,22 +22,32 @@ import utils.enums.Enums
 
 import java.time.LocalDate
 
-case class Business(
-    typeOfBusiness: TypeOfBusiness,
-    tradingName: Option[String],
-    firstAccountingPeriodStartDate: Option[LocalDate],
-    firstAccountingPeriodEndDate: Option[LocalDate],
-    latencyDetails: Option[LatencyDetails],
-    accountingType: Option[AccountingType],
-    commencementDate: Option[LocalDate],
-    cessationDate: Option[LocalDate],
-    businessAddressLineOne: Option[String],
-    businessAddressLineTwo: Option[String],
-    businessAddressLineThree: Option[String],
-    businessAddressLineFour: Option[String],
-    businessAddressPostcode: Option[String],
-    businessAddressCountryCode: Option[String]
-)
+case class Business(typeOfBusiness: TypeOfBusiness,
+                    tradingName: Option[String],
+                    firstAccountingPeriodStartDate: Option[LocalDate],
+                    firstAccountingPeriodEndDate: Option[LocalDate],
+                    latencyDetails: Option[LatencyDetails],
+                    accountingType: Option[AccountingType],
+                    commencementDate: Option[LocalDate],
+                    cessationDate: Option[LocalDate],
+                    businessAddressLineOne: Option[String],
```

Comment:
Have we thought about drawing out all the businessAddress fields into their own case class, as they are all strongly related fields? This would mean you won't need the `hasAnyBusinessAddressDetails` method.

Reply:
Yeah. I had a long discussion about how I really didn't like the way we modelled our API.

But the FE JSON itself is flat, so this case class is also flat. I strongly believe that we should have had two case classes to model the different APIs with their own readers/writers, instead of using mismatched JSON readers and writers on a single case class. But I didn't want to start refactoring bits I didn't properly understand.

A side effect of me not messing with the serializers/deserializers was that I left the case class alone.

Reply:
Mirroring the flatness makes sense - I didn't consider that.